agent/consul/acl.go

```diff
@@ -1087,6 +1094,32 @@ func (f *aclFilter) allowSession(node string) bool {
 	return f.authorizer.SessionRead(node)
 }
 
+// allowEvent is used to determine if a stream.Event is accessible for an ACL.
+func (f *aclFilter) allowEvent(event stream.Event) bool {
+	// Fast path if ACLs are not enabled
+	if f.authorizer == nil {
+		return true
+	}
+
+	allow := true
```

Comment:
Hmm I wonder if we could default to deny and then have an explicit `no ACLs required` sentinel value to encode in the protobuf so that we can reliably determine the difference between "no ACLs required" and "forgot to call the method that sets up the right ACLs in the event"?